### PR TITLE
IPA-EPN: Add design draft

### PR DESCRIPTION
**Pull request by fcami**
The design draft lists the user stories, implementation choices, implementation
details, limitations, and changes for the new EPN feature.

Fixes: https://pagure.io/freeipa/issue/3687
Signed-off-by: François Cami <fcami@redhat.com>

### CODE REVIEW

**Comment by rcritten** [comment]
Can you add Expired Password Notifications to the commit message:

The design draft lists the user stories, implementation choices, implementation
details, limitations, and changes for the new Expired Password Notifications (EPN) feature.

File: doc/designs/expiring-password-notification.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+A method to warn users via email that their IPA account password is about to expire.
+Ticket [link](https://pagure.io/freeipa/issue/3687).
+
+## User Stories
+
+[0] As an IPA user, I want to be notified by email and through the WebUI when my password is near its expiry date so that I change my password before it expires.
+
+Outcome: Users whose passwords are expiring receive an email.
+NB: The WebUI warning already exists. Document (ipapwdexpadvnotify) usage.
```
Comment (rcritten) [inline doc/designs/expiring-password-notification.md:15]: https://fedorahosted.org/freeipa/ticket/2625

File: doc/designs/expiring-password-notification.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+Outcome: Users receive notifications on an IPA-admin-specified list of days before their password expires.
+Supersedes [1].
+
+[3] As an IPA administrator, I want to be able to set smtp client parameters so that the notification emails are sent properly.
+
+Outcome: Emails are sent according to the SMTP parameters set in the configuration file.
+
+[4] As an IPA administrator, I do not want to notify users whose accounts are disabled because it makes no sense to.
+
+Outcome: If an account is disabled no emails are sent when the password is near expiry.
```
Comment (rcritten) [inline doc/designs/expiring-password-notification.md:32]: Worth specifically mentioning nsAccountLock or is that too much in the weeds?

File: doc/designs/expiring-password-notification.md (hunk 1 above)
Comment (fcami) [reply doc/designs/expiring-password-notification.md:15]: Added for completeness (it was already in the documentation section), thanks!

File: doc/designs/expiring-password-notification.md (hunk 2 above)
Comment (fcami) [reply doc/designs/expiring-password-notification.md:32]: Done, thanks!

**Comment by fcami** [comment]
I've changed the mocked-up json output. It is now closer to the goal.

**Comment by rcritten** [comment]
LGTM

**Comment by rcritten** [comment]
master:

* 4d2272f9667d35ef88f04153c49e0621c5d56d66 IPA-EPN: Add design draft

**Pull request closed without merging**